diff --git a/modules/store/src/Plugin/views/argument_default/CurrentStore.php b/modules/store/src/Plugin/views/argument_default/CurrentStore.php
index 9712d9b..3a1510e 100644
--- a/modules/store/src/Plugin/views/argument_default/CurrentStore.php
+++ b/modules/store/src/Plugin/views/argument_default/CurrentStore.php
@@ -83,7 +83,14 @@ class CurrentStore extends ArgumentDefaultPluginBase implements CacheableDepende
    * {@inheritdoc}
    */
   public function getCacheTags() {
-    return $this->currentStore->getStore()->getCacheTags();
+    $parent_cache_tags = parent::getCacheTags();
+    $store_cache_tags = [];
+
+    if ($store = $this->currentStore->getStore()) {
+      $store_cache_tags = Cache::mergeTags($store_cache_tags, $store->getCacheTags());
+    }
+
+    return Cache::mergeTags($parent_cache_tags, $store_cache_tags);
   }
 
 }
